src/bindings/js/node/src/async_infer_queue.cpp

```diff
@@ -0,0 +1,196 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "node/include/async_infer_queue.hpp"
+
+#include <condition_variable>
+#include <mutex>
+#include <queue>
+#include <string>
+#include <vector>
+
+#include "node/include/addon.hpp"
+#include "node/include/compiled_model.hpp"
+#include "node/include/errors.hpp"
+#include "node/include/helper.hpp"
+#include "node/include/infer_request.hpp"
+#include "node/include/node_output.hpp"
+#include "node/include/tensor.hpp"
+#include "node/include/type_validation.hpp"
+
+AsyncInferQueue::AsyncInferQueue(const Napi::CallbackInfo& info) : Napi::ObjectWrap<AsyncInferQueue>(info) {
+    const auto env = info.Env();
+    std::vector<std::string> allowed_signatures;
+
+    try {
+        const auto are_arguments_valid = ov::js::validate<CompiledModelWrap, int>(info, allowed_signatures);
+        OPENVINO_ASSERT(are_arguments_valid,
+                        "'AsyncInferQueue' constructor",
+                        ov::js::get_parameters_error_msg(info, allowed_signatures));
+
+        auto& compiled = Napi::ObjectWrap<CompiledModelWrap>::Unwrap(info[0].ToObject())->get_compiled_model();
+        size_t jobs = info[1].As<Napi::Number>().Int32Value();
+        m_tsfn = nullptr;
+
+        m_requests.reserve(jobs);
+        m_user_ids.reserve(jobs);
+        m_user_inputs.reserve(jobs);
+
+        for (size_t handle = 0; handle < jobs; handle++) {
+            m_requests.emplace_back(compiled.create_infer_request());
+            m_user_ids.push_back(std::make_pair(Napi::Reference<Napi::Object>::New(Napi::Object::New(env), 1),
+                                                Napi::Promise::Deferred::New(env)));
+            m_user_inputs.push_back(Napi::Reference<Napi::Object>::New(Napi::Object::New(env), 1));
+            m_idle_handles.push(handle);
+        }
+
+    } catch (const ov::Exception& e) {
+        reportError(env, e.what());
+    }
+}
+
+AsyncInferQueue::~AsyncInferQueue() {
+    m_requests.clear();
+    m_user_ids.clear();
+    m_user_inputs.clear();
+}
+
+void AsyncInferQueue::release() {
+    if (m_tsfn) {
+        const auto status = m_tsfn.Release();
+        if (status == napi_invalid_arg) {
+            OPENVINO_THROW("Failed to release AsyncInferQueue resources. "
+                           "ThreadSafeFunction is already released or not initialized.");
+        } else if (status != napi_ok) {
+            OPENVINO_THROW("Failed to release AsyncInferQueue resources.");
+        }
```

Comment:
```suggestion
        const auto status = m_tsfn.Release();
         OPENVINO_ASSERT(status == napi_invalid_arg, "Failed to release AsyncInferQueue resources. "
                           "ThreadSafeFunction is already released or not initialized.");
         OPENVINO_ASSERT(other_condition, "Failed to release AsyncInferQueue resources.");                           
```
Can be refactored to use this way